File: ios/Classes/LevelPlayNativeAdTemplateStyle.m
#import "LevelPlayNativeAdTemplateStyle.h"

@implementation LevelPlayNativeAdTemplateStyle

- (instancetype)initWithMainBackgroundColor:(NSNumber *)mainBackgroundColor
                        titleStyle:(LevelPlayNativeAdElementStyle *)titleStyle
                         bodyStyle:(LevelPlayNativeAdElementStyle *)bodyStyle
                   advertiserStyle:(LevelPlayNativeAdElementStyle *)advertiserStyle
                 callToActionStyle:(LevelPlayNativeAdElementStyle *)callToActionStyle {
    self = [super init];
    if (self) {
        _mainBackgroundColor = mainBackgroundColor;
        _titleStyle = titleStyle;
        _bodyStyle = bodyStyle;
        _advertiserStyle = advertiserStyle;
        _callToActionStyle = callToActionStyle;
    }
    return self;
}

- (id)copyWithZone:(NSZone *)zone {
    LevelPlayNativeAdTemplateStyle *copy = [[[self class] allocWithZone:zone] init];
    if (copy) {
        copy.mainBackgroundColor = [self.mainBackgroundColor copyWithZone:zone];
        copy.titleStyle = [self.titleStyle copyWithZone:zone];
        copy.bodyStyle = [self.bodyStyle copyWithZone:zone];
        copy.advertiserStyle = [self.advertiserStyle copyWithZone:zone];
        copy.callToActionStyle = [self.callToActionStyle copyWithZone:zone];
    }
    return copy;
}

@end

File: lib/src/models/level_play_native_ad_template_style.dart
import 'dart:ui';

import './level_play_native_ad_element_style.dart';

/// Class for styling native ad elements
class LevelPlayNativeAdTemplateStyle {
  final Color? mainBackgroundColor;
  final LevelPlayNativeAdElementStyle? titleStyle;
  final LevelPlayNativeAdElementStyle? bodyStyle;
  final LevelPlayNativeAdElementStyle? advertiserStyle;
  final LevelPlayNativeAdElementStyle? callToActionStyle;

  LevelPlayNativeAdTemplateStyle({
    this.mainBackgroundColor,
    this.titleStyle,
    this.bodyStyle,
    this.advertiserStyle,
    this.callToActionStyle,
  });

  Map<String, dynamic> toMap() {
    return <String, dynamic>{
      "mainBackgroundColor": mainBackgroundColor?.value,
      "titleStyle": titleStyle?.toMap(),
      "bodyStyle": bodyStyle?.toMap(),
      "advertiserStyle": advertiserStyle?.toMap(),
      "callToActionStyle": callToActionStyle?.toMap(),
    };
  }
}